graylog2-web-interface/src/components/common/EntityDataTable/contexts/DndStylesProvider.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import * as React from 'react';
+import { useMemo, useState } from 'react';
+import { useDndContext } from '@dnd-kit/core';
+
+import DndStylesContext from './DndStylesContext';
+
+const DndStylesProvider = ({ children }: React.PropsWithChildren) => {
```

Comment:
The Tanstack table example https://tanstack.com/table/latest/docs/framework/react/examples/column-dnd implements `useSortable` for every table header and table cell. 

This is problematic when using a `DragOverlay` which only includes the table header, because it will be displayed at the bottom of the column by default. It requires a custom `dnd-kit` modifier to change the initial position of the drag overlay. The modifier requires custom logic for keyboard and mouse interactions.

Instead we are now implementing `useSortable` only for the table headers. This required adding the `DndStylesProvider`, to properly style the table cells (transform and opacity) when dragging a column.
Before the table cells got this information from `useSortable`.